src/renderer/components/dock/logs.tsx

```diff
@@ -33,15 +32,33 @@ import { logStore } from "./log.store";
 import { LogSearch } from "./log-search";
 import { LogControls } from "./log-controls";
 import { LogTabData, logTabStore } from "./log-tab.store";
+import { podsStore } from "../+workloads-pods/pods.store";
+import { kubeWatchApi } from "../../../common/k8s-api/kube-watch-api";
+import { Spinner } from "../spinner";
+import { disposingReaction, comparer } from "../../utils";
+import type { Pod } from "../../../common/k8s-api/endpoints";
 
 interface Props {
   className?: string
   tab: DockTab
 }
 
+type GottenPods = {
+  pods: Pod[];
+  pod: Pod | undefined;
+} | {
+  pod: Pod | undefined;
+  pods?: undefined;
+};
```

Comment:
Why not just return empty array [] for `pods` when needed instead of this construction? To keep predictable return value.

Reply:
Because an empty array means that `workloadId` was set and because `[]` means no pods, where as `undefined` means did even try.

Reply:
Can you elaborate this a bit more? Can we convert this to 
```
pods: Pod[]
pod: Pod | undefined
```
?

Reply:
I guess it could be:

```
pods: Pod[] | undefined;
pod: Pod | undefined;
```

I wanted to capture in the type the two different states, but I see now that it isn't that much clearer.